src/components/CountdownCounter/CountdownCounter.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useContext, useEffect } from 'react';
+import { CircularProgressbar, buildStyles } from 'react-circular-progressbar';
+import { ThemeContext } from 'styled-components';
+import useInterval from 'src/hooks/useInterval';
+import { CountdownCounterProps } from './types';
+import 'react-circular-progressbar/dist/styles.css';
+
+const CountdownCounter: React.FC<CountdownCounterProps> = (props) => {
+  const [percentage, setPercentage] = React.useState<number>(props.value);
+  const theme = useContext(ThemeContext);
+
+  useEffect(() => {
+    if (percentage === props.minValue) {
+      props.onFinish && props.onFinish();
+    }
+  }, [percentage]);
```

Comment:
this useEffect should also contains minValue in dependancy array
```suggestion
  }, [percentage, minValue]);
```

Reply:
@bilalcorbacioglu this is useful when someone want to keep conditional minValue and on condition if minValue changes then useEffect should run again.